### ISSUE TITLE
Issue in `download_llama.py`

### ISSUE THREAD

**Issue by sakshiudeshi** [modal-labs/modal-examples#834]
I think I found a bug in the tutorials (https://modal.com/docs/examples/vllm_inference) that I'd like to report. 

In the `download_llama.py` file (https://github.com/modal-labs/modal-examples/blob/main/06_gpu_and_ml/llm-serving/download_llama.py)  lines 38 - 49 are as follows. 

```
 snapshot_download(
        model_name,
        local_dir=MODELS_DIR, ## Incorrect
        ignore_patterns=[
            "*.pt",
            "*.bin",
            "*.pth",
            "original/*",
        ],  # Ensure safetensors
        revision=model_revision,
        force_download=force_download,
    )
```
   
  The values of `MODELS_DIR` and `DEFAULT_NAME` are as follows (lines 6-8)
   
```
MODELS_DIR = "/llamas"

DEFAULT_NAME = "meta-llama/Meta-Llama-3.1-8B-Instruct"
``` 

Instead, I think lines 38 - 49 this should be 

```
 snapshot_download(
        model_name,
        local_dir=MODELS_DIR + "/" + DEFAULT_NAME,  ## Corrected
        ignore_patterns=[
            "*.pt",
            "*.bin",
            "*.pth",
            "original/*",
        ],  # Ensure safetensors
        revision=model_revision,
        force_download=force_download,
    )
```
    
    
    
This is because the `vllm_inference.py` (https://github.com/modal-labs/modal-examples/blob/main/06_gpu_and_ml/llm-serving/vllm_inference.py) expects both `MODELS_DIR` and `MODEL_NAME` in the path as seen in lines 143-149
    
    
    
```
    engine_args = AsyncEngineArgs(
        model=MODELS_DIR + "/" + MODEL_NAME,
        tensor_parallel_size=N_GPU,
        gpu_memory_utilization=0.90,
        max_model_len=8096,
        enforce_eager=False,  # capture the graph for faster inference, but slower cold starts (30s > 20s)
    )

```
    
Lines 40-41 in `vllm_inference.py`
    
    
    
  ```
  MODELS_DIR = "/llamas"
  MODEL_NAME = "meta-llama/Meta-Llama-3.1-8B-Instruct"
  ```

**Comment by charlesfrye** [comment]
thanks for reporting! resolved in #835

**Comment by c123ian** [comment]
Hey [https://github.com/modal-labs/modal-examples/pull/835](https://github.com/modal-labs/modal-examples/pull/835) worked for me thank you, but I now get a similar error but from the vLLM side, `loader.py` file, `_prepare_weights()` method cant find the model path.

error:

Runner failed with exception: RuntimeError('Cannot find any model weights with /llamas/c123ian/irish_eng_transl_2k_test_uuix')

```
WARNING 08-08 10:11:29 config.py:254] bitsandbytes quantization is not fully optimized yet. The speed can be slower than non-quantized models.
INFO 08-08 10:11:29 llm_engine.py:174] Initializing an LLM engine (v0.5.4) with config: model='/llamas/c123ian/irish_eng_transl_2k_test_uuix', speculative_config=None, tokenizer='/llamas/c123ian/irish_eng_transl_2k_test_uuix', skip_tokenizer_init=False, tokenizer_mode=auto, revision=None, rope_scaling=None, rope_theta=None, tokenizer_revision=None, trust_remote_code=False, dtype=torch.bfloat16, max_seq_len=4096, download_dir=None, load_format=LoadFormat.AUTO, tensor_parallel_size=1, pipeline_parallel_size=1, disable_custom_all_reduce=False, quantization=bitsandbytes, enforce_eager=True, kv_cache_dtype=auto, quantization_param_path=None, device_config=cuda, decoding_config=DecodingConfig(guided_decoding_backend='outlines'), observability_config=ObservabilityConfig(otlp_traces_endpoint=None), seed=0, served_model_name=/llamas/c123ian/irish_eng_transl_2k_test_uuix, use_v2_block_manager=False, enable_prefix_caching=False)
INFO 08-08 10:11:30 model_runner.py:720] Starting to load model /llamas/c123ian/irish_eng_transl_2k_test_uuix...
Traceback (most recent call last):
  File "/pkg/modal/_container_io_manager.py", line 462, in handle_user_exception
    yield
  File "/pkg/modal/_container_entrypoint.py", line 786, in main
    finalized_functions = service.get_finalized_functions(container_args.function_def, container_io_manager)
  File "/pkg/modal/_container_entrypoint.py", line 152, in get_finalized_functions
    web_callable = construct_webhook_callable(
  File "/pkg/modal/_container_entrypoint.py", line 68, in construct_webhook_callable
    return asgi_app_wrapper(user_defined_callable(), container_io_manager)
  File "/root/vllm_inference.py", line 149, in serve
    engine = AsyncLLMEngine.from_engine_args(
  File "/usr/local/lib/python3.10/site-packages/vllm/engine/async_llm_engine.py", line 471, in from_engine_args
    engine = cls(
  File "/usr/local/lib/python3.10/site-packages/vllm/engine/async_llm_engine.py", line 381, in __init__
    self.engine = self._init_engine(*args, **kwargs)
  File "/usr/local/lib/python3.10/site-packages/vllm/engine/async_llm_engine.py", line 552, in _init_engine
    return engine_class(*args, **kwargs)
  File "/usr/local/lib/python3.10/site-packages/vllm/engine/llm_engine.py", line 249, in __init__
    self.model_executor = executor_class(
  File "/usr/local/lib/python3.10/site-packages/vllm/executor/executor_base.py", line 47, in __init__
    self._init_executor()
  File "/usr/local/lib/python3.10/site-packages/vllm/executor/gpu_executor.py", line 36, in _init_executor
    self.driver_worker.load_model()
  File "/usr/local/lib/python3.10/site-packages/vllm/worker/worker.py", line 139, in load_model
    self.model_runner.load_model()
  File "/usr/local/lib/python3.10/site-packages/vllm/worker/model_runner.py", line 722, in load_model
    self.model = get_model(model_config=self.model_config,
  File "/usr/local/lib/python3.10/site-packages/vllm/model_executor/model_loader/__init__.py", line 21, in get_model
    return loader.load_model(model_config=model_config,
  File "/usr/local/lib/python3.10/site-packages/vllm/model_executor/model_loader/loader.py", line 328, in load_model
    self._get_weights_iterator(model_config.model,
  File "/usr/local/lib/python3.10/site-packages/vllm/model_executor/model_loader/loader.py", line 288, in _get_weights_iterator
    hf_folder, hf_weights_files, use_safetensors = self._prepare_weights(
  File "/usr/local/lib/python3.10/site-packages/vllm/model_executor/model_loader/loader.py", line 278, in _prepare_weights
    raise RuntimeError(
RuntimeError: Cannot find any model weights with `/llamas/c123ian/irish_eng_transl_2k_test_uuix`
Runner failed with exception: RuntimeError('Cannot find any model weights with `/llamas/c123ian/irish_eng_transl_2k_test_uuix`')
